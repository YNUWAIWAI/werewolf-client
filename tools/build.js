const fs = require('fs')
const path = require('path')
const postcss = require('postcss')
const nested = require('postcss-nested')
const autoprefixer = require('autoprefixer')
const glob = require('glob')
const webpack = require('webpack')
const webpackConfig = require('../webpack.config.js')

const mkdir = dir => {
  return new Promise((resolve, reject) => {
    fs.stat(path.dirname(dir), (err, stats) => {
      if (err) {
        if (err.code === 'ENOENT') {
          fs.mkdir(path.dirname(dir), err => {
            if (err) {
              reject(err)
            }
          })
        } else {
          reject(err)
        }
      }
      resolve()
    })
  })
}

const buildCSS = (src, dest) => {
  fs.readFile(src, (err, css) => {
    if (err) {
      throw err
    }
    postcss([ nested, autoprefixer ])
      .process(css, {
        from: src,
        to: dest
      })
      .then(result => {
        mkdir(dest)
          .then(() => {
            fs.writeFile(dest, result.css, err => {
              if (err) {
                throw err
              }
            })
          })
      })
      .catch(err => {
        console.log(err)
      })
  })
}

const buildHTML = (src, dest) => {
  fs.readFile(src, (err, data) => {
    if (err) {
      throw err
    }
    mkdir(dest)
      .then(() => {
        fs.writeFile(dest, data, err => {
          if (err) {
            throw err
          }
        })
      })
  })
}

const build = destDir => {
  const CSSfiles = glob.sync('src/**/*.css')
  const HTMLfiles = glob.sync('src/**/*.html')

  CSSfiles.forEach(file => {
    buildCSS(file, `${destDir}/stylesheets/${path.relative('src/styles', file)}`)
  })

  HTMLfiles.forEach(file => {
    buildHTML(file, `${destDir}/../app/views/${path.parse(file).name}.scala.html`)
  })
}

const compiler = webpack(webpackConfig)

if (process.argv[2] === '-w') {
  build('../public')
  fs.watch('./src', {recursive: true}, (eventType, filename) => {
    if (filename) {
      console.log(`${eventType}: ${filename}`)
      build('../public')
    }
  })
  compiler.watch({}, (err, stats) => {
    if (err) {
      throw err
    }
  })
} else if (process.argv[2] === '-d') {
  build('docs')
} else {
  build('../public')
  compiler.run((err, stats) => {
    if (err) {
      throw err
    }
    const info = stats.toJson()

    if (stats.hasErrors()) {
      console.error(info.errors)
    }

    if (stats.hasWarnings()) {
      console.warn(info.warnings)
    }
  })
}
