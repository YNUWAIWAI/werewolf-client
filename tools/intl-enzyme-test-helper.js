import {IntlProvider, intlShape} from 'react-intl'
import {mount, shallow} from 'enzyme'
import React from 'react'

export const initRenderer = (locale, messages) => {
  const intlProvider = new IntlProvider(
    {
      locale,
      messages
    },
    {}
  )
  const {intl} = intlProvider.getChildContext()
  const nodeWithIntlProp = node =>
    React.cloneElement(
      node,
      {
        intl
      }
    )

  return {
    mountWithIntl: (node, {context, childContextTypes, ... additionalOptions} = {}) =>
      mount(
        nodeWithIntlProp(node),
        {
          childContextTypes: {
            intl: intlShape,
            ... childContextTypes
          },
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      ),
    shallowWithIntl: (node, {context, ... additionalOptions} = {}) =>
      shallow(
        nodeWithIntlProp(node),
        {
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      )
  }
}

