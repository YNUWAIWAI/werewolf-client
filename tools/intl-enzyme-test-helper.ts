/* eslint no-unused-vars: 0 */
/// <reference types="lobby" />
import * as React from 'react'
import {IntlProvider, intlShape} from 'react-intl'
import {MountRendererProps, ShallowRendererProps, mount, shallow} from 'enzyme'

export const initRenderer = (locale: lobby.Language | village.Language, messages: {[key: string]: string}) => {
  const intlProvider = new IntlProvider(
    {
      locale,
      messages
    },
    {}
  )
  const {intl} = intlProvider.getChildContext()
  const nodeWithIntlProp = (node: React.ReactElement<any>) =>
    React.cloneElement(
      node,
      {
        intl
      }
    )

  return {
    mountWithIntl: (node: React.ReactElement<any>, {context, childContextTypes, ... additionalOptions}: MountRendererProps = {}) =>
      mount(
        nodeWithIntlProp(node),
        {
          childContextTypes: {
            intl: intlShape,
            ... childContextTypes
          },
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      ),
    shallowWithIntl: (node: React.ReactElement<any>, {context, ... additionalOptions}: ShallowRendererProps = {}) =>
      shallow(
        nodeWithIntlProp(node),
        {
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      )
  }
}

