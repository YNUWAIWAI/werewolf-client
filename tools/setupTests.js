const Adapter = require('enzyme-adapter-react-16')
const Enzyme = require('enzyme')
const {JSDOM} = require('jsdom')

Enzyme.configure({
  adapter: new Adapter()
})

const jsdom = new JSDOM(
  '<!doctype html><html><body></body></html>',
  {
    url: 'http://localhost'
  }
)
const {window} = jsdom

const copyProps = (src, target) => {
  Object.defineProperties(target, {
    ... Object.getOwnPropertyDescriptors(src),
    ... Object.getOwnPropertyDescriptors(target)
  })
}

global.window = window
global.document = window.document
global.navigator = {
  userAgent: 'node.js'
}
global.requestAnimationFrame = cb => setTimeout(cb, 0)
global.cancelAnimationFrame = id => {
  clearTimeout(id)
}
copyProps(window, global)
