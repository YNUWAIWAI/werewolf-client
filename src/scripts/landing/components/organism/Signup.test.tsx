import * as React from 'react'
import {
  Language,
  getMessages
} from '../../i18n'
import {Confirmation} from '../molecules/Confirmation'
import {Signup} from './Signup'
import {SubmitButton} from '../atoms/SubmitButton'
import {TextField} from '../molecules/TextField'
import {act} from 'react-dom/test-utils'
import {initRenderer} from '../../tools'

const {mountWithIntl} = initRenderer(Language.en, getMessages(Language.en))

test('render', () => {
  const wrapper = mountWithIntl(
    <Signup
      action="/signup"
      csrfToken="token"
      error={false}
      visible
    />
  )

  expect(wrapper.html()).toMatchSnapshot()
})
test('useEffect', () => {
  const wrapper = mountWithIntl(
    <Signup
      action="/signup"
      csrfToken="token"
      error={false}
      visible
    />
  )

  expect(wrapper.find(SubmitButton).props().disabled).toBe(true)
  act(() => {
    wrapper.find(TextField).at(0).props().handleValidityChange(true)
    wrapper.find(TextField).at(1).props().handleValidityChange(true)
    wrapper.find(TextField).at(2).props().handleValidityChange(true)
    wrapper.find(Confirmation).props().handlePrivacyCheckChange(true)
    wrapper.find(Confirmation).props().handleTermsCheckChange(true)
  })
  wrapper.update()
  expect(wrapper.find(SubmitButton).props().disabled).toBe(false)
})
