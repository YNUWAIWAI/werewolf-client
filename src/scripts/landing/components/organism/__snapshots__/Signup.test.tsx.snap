// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `"<form action=\\"/signup\\" class=\\"la--form \\" method=\\"POST\\"><div class=\\"la--form--field\\"><label class=\\"la--form--label\\" for=\\"signup-name\\">Name</label><input autocomplete=\\"off\\" class=\\"la--form--input\\" id=\\"signup-name\\" minlength=\\"5\\" name=\\"name\\" pattern=\\"^[A-Za-z](?:[!-~]| [!-~]){4,14}$\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"la--form--field-description\\">5-15 characters (except space). Disallow multi-space.</div><div class=\\"la--form--field\\"><label class=\\"la--form--label\\" for=\\"signup-email\\">Email address</label><input autocomplete=\\"off\\" class=\\"la--form--input\\" id=\\"signup-email\\" minlength=\\"6\\" name=\\"email\\" pattern=\\"[^\\\\s]*@[^\\\\s]*\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"la--form--field-description\\">6-256 characters. Latin-1.</div><div class=\\"la--form--field\\"><label class=\\"la--form--label\\" for=\\"signup-password\\">Password</label><input autocomplete=\\"off\\" class=\\"la--form--input\\" id=\\"signup-password\\" minlength=\\"8\\" name=\\"password\\" required=\\"\\" type=\\"password\\" value=\\"\\"></div><div class=\\"la--form--field-description\\">8-128 characters.</div><div class=\\"la--form--confirmation\\"><span aria-checked=\\"false\\" class=\\"la--checkbox\\" role=\\"checkbox\\" tabindex=\\"0\\"><svg class=\\"checkbox blank\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path></svg></span><span>Please agree to our <a class=\\"la--form--link\\" href=\\"https://werewolfguide.netlify.app/docs/en/privacy\\" rel=\\"noreferrer noopener\\" target=\\"_blank\\">Privacy Policy</a> before signing up.</span><span aria-checked=\\"false\\" class=\\"la--checkbox\\" role=\\"checkbox\\" tabindex=\\"0\\"><svg class=\\"checkbox blank\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path></svg></span><span>Please agree to our <a class=\\"la--form--link\\" href=\\"https://werewolfguide.netlify.app/docs/en/terms\\" rel=\\"noreferrer noopener\\" target=\\"_blank\\">Terms of Use</a> before signing up.</span></div><button class=\\"la--form--button\\" disabled=\\"\\" type=\\"submit\\">Signup</button><input name=\\"csrfToken\\" type=\\"hidden\\" value=\\"token\\"></form>"`;
